core/lsp/TypecheckEpochManager.cc

```diff
@@ -0,0 +1,116 @@
+#include "core/lsp/TypecheckEpochManager.h"
+#include "core/lsp/PreemptionTaskManager.h"
+
+using namespace std;
+namespace sorbet::core::lsp {
+
+void TypecheckEpochManager::assertConsistentThread(optional<thread::id> &expectedThreadId, string_view method,
```

Comment:
We don't know all of the thread IDs at construction time, so this assertion primarily asserts that we call specific routines from a consistent thread ID.